simtools/data_model/validate_data.py

```diff
@@ -142,10 +144,12 @@ def _validate_data_dict(self):
         unit_as_list = [None if unit == "null" else unit for unit in unit_as_list]
         for index, (value, unit) in enumerate(zip(value_as_list, unit_as_list)):
             if self._get_data_description(index).get("type", None) == "dict":
-                self._logger.debug(f"Skipping validation of dict type for entry {index}")
+                self._validate_data_dict_using_json_schema(
```

Comment:
Codecov complains this line is not covered by tests.

Reply:
Yes, I consciously didn't add it as I either would have to add a complex example with data and jsonschema or define at least 7 mocker patches. I don't want to do the latter and as this is kind of a trivial line I think it can be such.

(and I know that this is actually a hint that this function is too complex; suprised that sonar is not complaining)

Reply:
I am really hesitant to be honest. I agree that the line itself is not important cause the function it calls is tested elsewhere.  On the other hand, it definitely points to the fact that this function would be better if it were simplified. I leave it to you to decide then. If you think refactoring is worth it, good. Otherwise I approve already and we can ignore this warniing.

Reply:
Will refactoring it, but not as part of this pull request: #1193